src/identity-providers/add/DetailSettings.tsx

```diff
@@ -96,11 +105,15 @@ export const DetailSettings = () => {
   const { realm } = useRealm();
 
   useFetch(
-    () => adminClient.identityProviders.findOne({ alias: alias }),
-    (provider) => {
-      if (provider) {
-        setProvider(provider);
-        Object.entries(provider).map((entry) => setValue(entry[0], entry[1]));
+    () =>
+      Promise.resolve([
+        adminClient.identityProviders.findOne({ alias: alias }),
+      ]),
+    async ([fetchedProvider]) => {
+      if (fetchedProvider) {
+        Object.entries(fetchedProvider).map(([key, value]) =>
+          setValue(key, value)
+        );
       }
```

Comment:
This can be reduced/improved a bit by making the following changes:

1. There is no need to wrap the call in `Promise.resolve()` as it already returns a Promise
2. There is no need to make the callback `async` as there is no `await` used inside of the function.
3. There is no need to destructure the result as it's only a single value.
4. There is no need to check the value of the result before assignment (according to the return type it's value is always present). 
5. `forEach()` should be used instead of `.map()` as there is no resulting value from the map operation.

With these changes you'll end up with the following:
```suggestion
    () => adminClient.identityProviders.findOne({ alias: alias }),
    (provider) => {
      Object.entries(provider).forEach(([key, value]) => setValue(key, value));
    },
```